packages/grid/_modules_/grid/hooks/features/editRows/useGridEditRows.ts

```diff
@@ -106,24 +100,28 @@ export function useGridEditRows(
     }
   };
 
-  const handleCellFocusIn = React.useCallback((params) => {
-    nextFocusedCell.current = params;
-  }, []);
+  const handleCellFocusIn = React.useCallback<GridEventListener<GridEvents.cellFocusIn>>(
+    (params) => {
+      nextFocusedCell.current = params;
+    },
+    [],
+  );
 
-  const handleCellFocusOut = useEventCallback(
-    (params: GridCellParams, event: MouseEvent | React.SyntheticEvent) => {
+  const handleCellFocusOut: GridEventListener<GridEvents.cellFocusOut> = useEventCallback(
+    (params, event) => {
       commitPropsAndExit(params, event);
     },
   );
 
-  const handleColumnHeaderDragStart = useEventCallback((nativeEvent: any) => {
-    const { cell } = apiRef.current.state.focus;
-    if (!cell) {
-      return;
-    }
-    const params = apiRef.current.getCellParams(cell.id, cell.field);
-    commitPropsAndExit(params, nativeEvent);
-  });
+  const handleColumnHeaderDragStart: GridEventListener<GridEvents.columnHeaderDragEnter> =
+    useEventCallback(() => {
+      const { cell } = apiRef.current.state.focus;
+      if (!cell) {
+        return;
+      }
+      const params = apiRef.current.getCellParams(cell.id, cell.field);
+      commitPropsAndExit(params, {});
```

Comment:
@m4theushw I have an interesting case here.

From what I understand, you mis-used `useEventCallback`. The event is on the 2nd argument since it is still one of our listeners. So we were passing the `GridColumnHeaderParams` as event to `commitPropsAndExit`.

I did a 1st fix by passing the actual event  

```tsx
    useEventCallback((dragParams, nativeEvent) => {
      const { cell } = apiRef.current.state.focus;
      if (!cell) {
        return;
      }
      const params = apiRef.current.getCellParams(cell.id, cell.field);
      commitPropsAndExit(params, nativeEvent);
```

But it seems that the event has `isPropagationStopped() === true` in some cases.
So the event in `commitPropsAndExit` is swallowed and we never leave the edit mode.

To bypass this issue I stopped passing the event to `commitPropsAndExit`, which works but is radical.

Reply:
> From what I understand, you mis-used useEventCallback. The event is on the 2nd argument since it is still one of our listeners.

Yes, it was wrong.

> But it seems that the event has isPropagationStopped() === true in some cases.

Do you have an example? We should avoid stopping the propagation of events because it prevent users from listening to them. I could only find https://github.com/mui-org/material-ui-x/blob/e126eb9179c2643fb2bc98619909273e54bdccc5/packages/grid/_modules_/grid/hooks/features/columnReorder/useGridColumnReorder.tsx#L89

However, calling `event.stopPropagation()` above only stops further listeners that are binded to different events. The listener from the editing hook should still be called.

Reply:
In my PR if you pass the `nativeEvent` on the `commitPropsAndExit` of line 123 like so:

```tsx
  const handleColumnHeaderDragStart: GridEventListener<GridEvents.columnHeaderDragEnter> =
    useEventCallback((_, nativeEvent) => {
      const { cell } = apiRef.current.state.focus;
      if (!cell) {
        return;
      }
      const params = apiRef.current.getCellParams(cell.id, cell.field);
      commitPropsAndExit(params, nativeEvent);
    });
```

You have `should save changes when a column header is dragged` failing
Because the cell is never leaving edit mode.

The guilty `stopPropagation` is here: https://github.com/mui-org/material-ui-x/blob/master/packages/grid/_modules_/grid/hooks/features/columnReorder/useGridColumnReorder.tsx#L89

Reply:
Now I understood. It's calling `commitPropsAndExit` but `cellEditStop` is never published because its event has propagation stopped. Not passing an event seems to be the only alternative. In v6 we should remove the `stopPropagation` and handle the integration between React DnD differentely.